packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -85,123 +109,506 @@ const propTypes = {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.MONTH) {
+          dayInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateOrder === DateUtil.dateOrder.DMY) {
+      if (inputValue.length === 2) {
+        if (type === DateUtil.inputType.DAY) {
+          monthInputRef.current.focus();
+        } else {
+          yearInputRef.current.focus();
+        }
+      }
+    } else if (dateOrder === DateUtil.dateOrder.YMD) {
+      if (inputValue.length === 4) {
+        monthInputRef.current.focus();
+      } else if (inputValue.length === 2 && type === DateUtil.inputType.MONTH) {
+        dayInputRef.current.focus();
+      }
+    }
+  };
+
+  /**
+   * Sets state for the day, month, and year.
+   * @param {string} inputValue - The input value to set in state.
+   * @param {number} type - The inputType (day, month, or year).
+   */
+  const setDate = (inputValue, type) => {
+    if (type === DateUtil.inputType.DAY) {
+      dateDispatch({ day: inputValue });
+    } else if (type === DateUtil.inputType.MONTH) {
+      dateDispatch({ month: inputValue });
+    } else {
+      dateDispatch({ year: inputValue });
+    }
+
+    moveFocusOnChange(inputValue, type, dateFormatOrder);
+  };
 
-    this.uuid = uuidv4();
+  const handleDateChange = (event, inputValue, type) => {
+    /**
+     * Sets the day, month and year based on input values, formats them
+     * based on the date format variant, and passes the formatted date to onChange.
+     */
+    const day = type === DateUtil.inputType.DAY ? inputValue : date.day;
+    const month = type === DateUtil.inputType.MONTH ? inputValue : date.month;
+    const year = type === DateUtil.inputType.YEAR ? inputValue : date.year;
+
+    if (type === DateUtil.inputType.DAY) {
+      setDayInitialFocused(false);
+    } else if (type === DateUtil.inputType.MONTH) {
+      setMonthInitialFocused(false);
+    } else {
+      setYearInitialFocused(false);
+    }
 
-    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
-    this.handleOnChange = this.handleOnChange.bind(this);
-    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
-  }
+    let inputDate;
+    let formattedDate;
+    if (day.length === 2 && month.length === 2 && year.length === 4) {
+      inputDate = DateUtil.convertToISO8601(`${year}-${month}-${day}`, DateUtil.ISO_EXTENDED_DATE_FORMAT);
+      formattedDate = DateUtil.strictFormatISODate(inputDate, momentDateFormat);
+    }
 
-  componentDidUpdate(prevProps) {
-    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
-      this.props.onClick();
+    if (onChange) {
+      if (DateUtil.isValidDate(formattedDate, momentDateFormat)) {
+        onChange(event, formattedDate);
+      } else {
+        let dateString;
+        if (dateFormatOrder === DateUtil.dateOrder.MDY) {
+          dateString = month + day + year;
```

Comment:
`OnChange` in terra-time-input and terra-date-input also returns the separator (i.e. `:` for time and `-` for date). The following entry returns `052020`. It's hard to consumers to parse if needed. Should it also return the separators as in `05//2020`. This means we need to get the separator (i.e. `/`; `-`; or `.`) based on the locale.

![Screen Shot 2020-10-06 at 2 22 41 PM](https://user-images.githubusercontent.com/22618655/95250572-ebcb4200-07df-11eb-8497-acf76e56084e.png)

Reply:
Looks like this still doesn't have the separators.

Reply:
Addressed here : https://github.com/cerner/terra-framework/pull/1208/commits/a3ba740adba9699efbe60c5cfee0e655ce7f2489